BCStudentSoftwareDevTeam/lsf
# Create simple and advanced views for supervisor portal

## Issue
Fixes issue #441 

We created a new supervisor portal, and that is great, but we took it from a user friendly (albeit a very crappily designed) experience to a big table with a ton of info. We should give them the option to view in a clearer "simpler" view while maintaining all the same functionality and features of the more advanced view.

## Changes
- Created a button in the top right that allows you to toggle between a simple and advanced view.
- Created a new datatable that renders when Switch To Simple View is toggled that limits the information and presents it in a slightly more user friendly way.
- Added new logic in Flask to handle the rendering of a simple view.
- Modified the cookie handling logic so when the data is pulled in from the session cookies it renders the correct view.

## To Test
- Checkout `441simple_view`
- Run `./database/reset_database.sh from-backup` and `export FLASK_ENV=staging`
- `flask run`
- Upon load you should see the supervisor portal.
- Execute a search and verify that it displays properly on the simple view.
- Try sorting and make sure you can only sort by columns and fields that are directly visible to the user.
- Click "Switch to Advanced View" and verify that the data is the same but just displayed differently.
- Try sorting and make sure you can sort by all columns and fields
- Execute a search with advanced view and close the window
- Reload the window and make sure that all data is automatically loaded in properly.
